src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/seed/cas1/ApprovedPremisesRoomsSeedFromXLSXJob.kt

```diff
@@ -0,0 +1,210 @@
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.BedEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.BedRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.CharacteristicEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.CharacteristicRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.PremisesEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.PremisesRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.RoomEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.RoomRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.seed.ExcelSeedJob
+import java.util.UUID
+
+class SiteSurveyImportException(message: String) : Exception(message)
+
+@Component
+@Suppress("LongParameterList")
+class ApprovedPremisesRoomsSeedFromXLSXJob(
+  private val premisesRepository: PremisesRepository,
+  private val roomRepository: RoomRepository,
+  private val bedRepository: BedRepository,
+  private val siteSurvey: SiteSurvey,
+) : ExcelSeedJob() {
+  private val log = LoggerFactory.getLogger(this::class.java)
+  override fun processDataFrame(dataFrame: DataFrame<*>, premisesId: UUID) {
+    findExistingPremisesOrThrow(premisesId)
+
+    // build valid lists for rooms, characteristics and beds
+    var rooms = buildRooms(dataFrame, premisesId)
+    val characteristics = buildCharacteristics(dataFrame)
+
+    // add the characteristics to the rooms if they exist
+    rooms.forEach {
+      characteristics[it.code]?.let { roomCharacteristics -> it.characteristics.addAll(roomCharacteristics.toList()) }
+    }
+
+    // update the rooms
+    rooms = createOrUpdateRooms(rooms)
+
+    val beds = buildBeds(dataFrame, rooms)
+    createBedsIfNotExist(beds)
+  }
+
+  private fun buildRooms(dataFrame: DataFrame<*>, premisesId: UUID): MutableList<RoomEntity> {
+    val rooms = mutableListOf<RoomEntity>()
+
+    for (i in 1..<dataFrame.columnsCount()) {
+      val roomAnswers = dataFrame.getColumn(i)
+      val room = createRoom(premisesId, roomAnswers.name, roomAnswers[0].toString())
+
+      rooms.add(room)
+    }
+    return rooms
+  }
+
+  private fun buildCharacteristics(dataFrame: DataFrame<*>): MutableMap<String, MutableList<CharacteristicEntity>> {
+    var premisesCharacteristics = mutableMapOf<String, MutableList<CharacteristicEntity>>()
+
+    siteSurvey.questionToCharacterEntityMapping.forEach { (question, characteristic) ->
+      val rowId = dataFrame.getColumn(0).values().indexOf(question)
+
+      if (rowId == -1) throw SiteSurveyImportException("Characteristic question '$question' not found on sheet Sheet3.")
+
+      for (colId in 1..<dataFrame.columnsCount()) {
+        val roomCode = dataFrame.getColumn(colId).name
+        val answer = dataFrame[rowId][colId].toString().trim()
+
+        if (answer.equals("yes", true)) {
```

Comment:
could you please adding a parameter name for the second argument (ignore) for readibility

Reply:
Added parameter name.